core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -344,10 +344,9 @@ public void cancel(@Nullable String message, @Nullable Throwable cause) {
           status = status.withCause(cause);
         }
         if (message == null && cause == null) {
-          // TODO(zhangkun83): log a warning with this exception once cancel() has been deleted from
-          // ClientCall.
           status = status.withCause(
               new CancellationException("Client called cancel() without any detail"));
+          log.log(Level.WARNING, "Suboptimal use of cancel", status.getCause());
```

Comment:
"Suboptimal" doesn't really tell you what is wrong or how to fix it. Maybe "Cancelled without a message or cause"? We may also want to do that check at the very beginning of the method to avoid racing with whether cancel has already been called.